.github/problem-matchers/clang-errors-warnings.json

```diff
@@ -0,0 +1,17 @@
+{
+    "problemMatcher": [
+        {
+            "owner": "clang-errors-warnings",
+            "pattern": [
+                {
+                    "regexp": "^(?:/builds/)?(.*):(\\d+):(\\d+):\\s+(?:fatal\\s+)?(warning|error):\\s+(.*)$",
```

Comment:
Perhaps we want to match uppercase `WARNING` as well (ie. from modpost).

Is it intentional that some of these groups use matching, and others don't (`?:`)?

What is `/builds/`? Does that string appear in kernel build logs?

Reply:
> Perhaps we want to match uppercase `WARNING` as well (ie. from modpost).

modpost warning are not the same pattern though so they won't match. I do eventually want to make problem matchers for all of these types of problems though.


> Is it intentional that some of these groups use matching, and others don't (`?:`)?

Yes, I believe so.

> What is `/builds/`? Does that string appear in kernel build logs?

Our source is mounted at `/builds/linux` within TuxSuite so that is the base of our warnings.